homeassistant/components/sensor/discogs.py

```diff
@@ -40,32 +65,45 @@
     """Set up the Discogs sensor."""
     import discogs_client
 
-    name = config.get(CONF_NAME)
-    token = config.get(CONF_TOKEN)
+    token = config[CONF_TOKEN]
+    name = config[CONF_NAME]
 
     try:
-        discogs = discogs_client.Client(SERVER_SOFTWARE, user_token=token)
-        identity = discogs.identity()
+        discogs_client = discogs_client.Client(
+            SERVER_SOFTWARE, user_token=token)
+
+        discogs_data = {
+            'user': discogs_client.identity().name,
```

Comment:
This is doing non asyncio I/O in a coroutine. That's not allowed. If the library doesn't support asyncio, it's better to use the home assistant sync api. Change to `setup_platform`.

Reply:
Aha I was unsure about when to use async methods, so you only want that when the library is using async under the hood?

Reply:
Usually that's the way to go, since it usually makes things simpler. There are exceptions.